lib/authorization.js

```diff
@@ -14,32 +14,35 @@ var authorizeWithRefreshSAS = function authorizeWithRefreshSAS(method, path, que
     debug("Refreshing shared-access-signature");
     // Avoid refreshing more than once
     this._nextSASRefresh = 0;
-    // Refresh SAS
-    this._sas = Promise.resolve(this.options.sas());
-    // Update _nextSASRefresh when the SAS has been refreshed
-    this._sas.then(function(sas) {
-      sas = querystring.parse(sas);
-      // Find next sas refresh time
-      self._nextSASRefresh = (
-        new Date(sas.se).getTime() - self.options.minSASAuthExpiry
-      );
-      debug("Refreshed shared-access-signature, will refresh in %s ms",
-        self._nextSASRefresh);
-      // Throw an error if the signature expiration comes too soon
-      if (Date.now() > self._nextSASRefresh) {
-        throw new Error("Refreshed SAS, but got a Shared-Access-Signature " +
-          "that expires less than options.minSASAuthExpiry " +
-          "from now, signature expiry: " + sas.se);
-      }
-    }).catch(function(err) {
-      // If we have an error freshing SAS that's bad and we'll emit it; for most
-      // apps it's probably best to ignore this error and just crash.
-      self.emit('error', err);
-    });
+    // Refresh SAS and update _nextSASRefresh when the SAS has been refreshed
+    this._sas = Promise.resolve(this.options.sas())
+      .then(function(sas) {
```

Comment:
This changes semantics from before... Did you notice that?

I suspect I intentionally didn't do: `this._sas = this._sas.then(...` for a reason.

Reply:
Yes, the old semantics were wrong :)

A promise doesn't "mutate" when you call its `.then` method, so the old code was creating a new promise and dropping it; when that promise was rejected, we got an unhandled rejection error.

This behavior of Promises is, I think, undefined, which is why it changed between Node 4 and Node 6.

Reply:
It has a `.catch(...` so it couldn't be unhandled!

And yes, I'm pretty sure the non-mutation was intentional.

> This behavior of Promises is, I think, undefined, which is why it changed between Node 4 and Node 6.

What was undefined? I see nothing of the sorts :)
Calling `.then(..` multiple times on the same promise is well-defined. And there is nothing wrong with it.

Reply:
Ah, I see that it changed the value of that Promise.  That was my mistake (and part of why this was not passing tests..)

Reply:
It was unhandled since it just called `.emit` on an object that is not an EventEmitter.  I assume thta was a leftover from some previous revision where these things did emit events?